### PR TITLE
add pin completion listener

### PR DESCRIPTION
**Pull request by CDRussell**
gives a way to specify a callback which will be given the pin as soon as all of the digits have been entered.

### CODE REVIEW

**Comment by CDRussell** [comment]
merged upstream changes to resolve merge conflicts

**Comment by CDRussell** [comment]
Do you think you'll get a chance to review this @Philio ?

**Comment by Philio** [comment]
Oops sorry @CDRussell, I'll check it now

File: pinentry/src/main/java/me/philio/pinentry/PinEntryView.java [hunk 1]
```diff
@@ -316,6 +333,7 @@ private void addViews() {
         editText.setFilters(new InputFilter[]{new InputFilter.LengthFilter(digits)});
         editText.setInputType(inputType);
         editText.setImeOptions(EditorInfo.IME_FLAG_NO_EXTRACT_UI);
+        editText.setPadding(editText.getPaddingLeft(), editText.getPaddingTop(), editText.getPaddingRight(), 100);
```
Comment (Philio) [inline pinentry/src/main/java/me/philio/pinentry/PinEntryView.java]: Why does the hidden `EditText` need padding?

Comment (CDRussell) [inline pinentry/src/main/java/me/philio/pinentry/PinEntryView.java]: Hi @Philio, good catch; this does look like a mistake. I've removed this change.

**Comment by ryanjohn1** [comment]
Hi, is this going to make it into a release soon? It's the exact functionality we're looking for in our app but I notice it isn't in 1.0.6

**Pull request merged**